nuxeo-core/nuxeo-core-storage-dbs/src/main/java/org/nuxeo/ecm/core/storage/dbs/DBSTransactionState.java

```diff
@@ -296,6 +301,65 @@ public boolean hasChild(String parentId, String name) {
         }
         // fetch from repository
         List<State> states = repository.queryKeyValue(KEY_PARENT_ID, parentId, seen);
+        return getDbsDocumentStates(docStates, states);
+    }
+
+    public List<DBSDocumentState> getRegularChildrenStates(String parentId) {
+        List<DBSDocumentState> docStates = new LinkedList<>();
+        Set<String> seen = new HashSet<>();
+        Set<String> specialChildrenTypeNames = getSpecialChildrenTypesNamesSet();
+        // check transient state
+        for (DBSDocumentState docState : transientStates.values()) {
+            if (!parentId.equals(docState.getParentId())) {
+                continue;
+            }
+            if (specialChildrenTypeNames.contains(docState.getPrimaryType())) {
+                continue;
+            }
+            docStates.add(docState);
+            seen.add(docState.getId());
+        }
+        // fetch from repository
+        Map<String, List<String>> filter = getSpecialChildrenFilter(NIN, specialChildrenTypeNames);
+        List<State> states = repository.queryKeyValue(KEY_PARENT_ID, parentId, KEY_PRIMARY_TYPE, filter, seen);
```

Comment:
I'm not seeing the MongoDB or Mem implementation of this new API with a `filter`...

Reply:
The filter is passed as the second key, object of the already existing `org.nuxeo.ecm.core.storage.dbs.DBSRepository.queryKeyValue(String, Object, String, Object, Set<String>)`
So we don't need a new api

Reply:
Following to our live review, a new api expliciting the operator will be created